civis/parallel.py

```diff
@@ -161,34 +166,34 @@ def infer_backend_factory(required_resources=None,
     # Update arguments with input
     state.arguments.update(arguments or {})
 
-    return make_backend_factory(docker_image_name=state.docker_image_name,
-                                docker_image_tag=state.docker_image_tag,
-                                repo_http_uri=state.repo_http_uri,
-                                repo_ref=state.repo_ref,
-                                required_resources=state.required_resources,
+    # Set defaults on other keyword arguments with
+    # values from the current script
+    for key in ['docker_image_name', 'docker_image_tag', 'repo_http_uri',
+                'repo_ref', 'remote_host_credential_id', 'git_credential_id',
+                'cancel_timeout', 'time_zone']:
+        kwargs.setdefault(key, state[key])
+
+    return make_backend_factory(required_resources=state.required_resources,
                                 params=state.params,
                                 arguments=state.arguments,
                                 client=client,
                                 polling_interval=polling_interval,
                                 setup_cmd=setup_cmd,
                                 max_submit_retries=max_submit_retries,
                                 max_job_retries=max_job_retries,
-                                hidden=hidden)
+                                hidden=hidden,
+                                **kwargs)
 
 
 def make_backend_factory(docker_image_name="civisanalytics/datascience-python",
                          docker_image_tag="latest",
-                         repo_http_uri=None,
-                         repo_ref=None,
-                         required_resources=None,
-                         params=None,
-                         arguments=None,
```

Comment:
Mmmmk. I am not 100% sure this is the best thing to do here, but let's go with it for now. We can deprecate later.